### PR TITLE
Properly handle absence of the Coq-LSP in UI and allow modifying its standard location

### PR DESCRIPTION
**Pull request by K-dizzled**
YouTrack ISSUE: [PLAN-127](https://youtrack.jetbrains.com/issue/PLAN-127) 

Main internal change: CoqLspClient now has a private constructor and a fabric async method for construction
Reason for the change: The async `start` method of the BaseLanguageClient class was ran in the constructor and was not awaited. Thus, the only way to figure out that the client is incorrectly created/failed to start -- was afterwards, after calling its methods. As constructors in typescript are not allowed to be made async, I decided on this solution.

### CODE REVIEW

File: src/coqLsp/coqLspClient.ts
Comment (K-dizzled) [inline src/coqLsp/coqLspClient.ts:1]: Main change

File: src/coqLsp/coqLspConnector.ts
Comment (K-dizzled) [inline src/coqLsp/coqLspConnector.ts:1]: Error now throwed out of the method, when started incorrectly. Not swallowed

File: src/coqLsp/coqLspTypes.ts [hunk 1]
```diff
@@ -156,3 +156,13 @@ export class CoqLspError extends Error {
         this.name = "CoqLspError";
     }
 }
+
+export class CoqLspStartupError extends Error {
```
Comment (GlebSolovev) [inline src/coqLsp/coqLspTypes.ts:160]: Maybe it's reasonable to make `CoqLspStartup` extend `CoqLspError` as a specific case of Coq LSP failure? Or if it does not seem right, I'd suggest renaming `CoqLspError` to something more specific of what it should actually mean.

File: src/coqLsp/coqLspTypes.ts [hunk 2]
```diff
@@ -156,3 +156,13 @@ export class CoqLspError extends Error {
...
+    constructor(
+        message: string,
+        public readonly path: string
```
Comment (GlebSolovev) [inline src/coqLsp/coqLspTypes.ts:163]: `public` is redundant here (and according to our unwritten code style rules, we usually omit it)

File: src/coqLsp/coqLspTypes.ts [hunk 3]
```diff
@@ -156,3 +156,13 @@ export class CoqLspError extends Error {
...
+    ) {
+        super(message);
+        this.name = "CoqLspStartupError";
```
Comment (GlebSolovev) [inline src/coqLsp/coqLspTypes.ts:166]: I'm not an expert in creating custom errors, so could you please share the idea behind this line? Also, I [found on the Internet](https://stackoverflow.com/a/41429145) that extending `Error` might actually break `instanceof` and related things: but there the suggested approach was to update the prototype
```TypeScript
Object.setPrototypeOf(this, CoqLspStartupError.prototype);
```

**Review by GlebSolovev — CHANGES_REQUESTED** [review]
Looks very lovely to me 😍 

Nice idea of fixing the problem and making the usability improvement together, clear README updates, and small and actually needed refactor 🤝 Cool!

File: src/coqLsp/coqLspTypes.ts (hunk 3 above)
Comment (K-dizzled) [reply src/coqLsp/coqLspTypes.ts:166]: Error names are used for example in our logging. E.g. `src/llm/llmServices/utils/generationsLogger/generationsLogger.ts`

Comment (K-dizzled) [reply src/coqLsp/coqLspTypes.ts:166]: This is done to override parent's name.

Comment (K-dizzled) [reply src/coqLsp/coqLspTypes.ts:166]: I made a series of experiments that showed no benefits of using `Object.setPrototypeOf(this, CoqLspStartupError.prototype);` and instanceof show no reasons to worry. Everything works fine, at least for reasonable typescript versions.

File: src/coqLsp/coqLspTypes.ts (hunk 1 above)
Comment (K-dizzled) [reply src/coqLsp/coqLspTypes.ts:160]: Done. Also moved other extensions of the CoqLspError to the same location.

**Pull request merged**